src/lib/data/source-text.js

```diff
@@ -76,6 +78,23 @@ export default class SourceText {
 
     this.sourceType = docSource.sourceType ? docSource.sourceType : this.sourceType
     this.tokenization = Object.assign({}, docSource.tokenization)
+
+    if (this.text.length === 0) {
+      this.detectedLang = false
+    }
+  }
+
+  updateDetectedLang (langData) {
+    this.sourceType = langData.sourceType
+    if (langData.lang) {
+      this.lang = langData.lang
+      this.direction = langData.direction
+    }
+    this.detectedLang = true
+  }
+
+  get readyForLangDetection () {
+    return this.text && (this.text.length > 5) && !this.detectedLang

```

Comment:
These are minimal conditions for language detection

Reply:
Am I understanding the logic of this code correctly?

1. Once the text is longer than 5 characters, the language will be detected
2. The language will not be detected again unless the text length is 0

This seems like it could lead to problems for users who are typing text in the box. In this case, wouldn't we want to check the language when there is more data available? Or is the expected workflow that users change the language manually when typing text?

Reply:
The logic here is the following:

- the event that will update current text length - is `blur`. It is done to reduce amount of text updates due to the fact, that text could be rather big.
- so the text would be updated and length would be calculated inside this text only after one of the follwoing events:
   - user unfocus the textarea
   - user uploaded the text from external resource
   
After a user removed the focus - the check would be done, and a user will be given an option to change language/direction and sourceType manually.

I set a limit to the 5 characters because if there are less characters - it is too few for lnguage detection.

@zfletch , in this case do you see additional risks?

Reply:
Thanks for the explanation! If the language detection code only runs on `blur`, I don't think there is much danger.